pantheon-bundle/src/main/java/com/redhat/pantheon/model/document/DocumentMetadata.java

```diff
@@ -37,6 +37,9 @@
     @Named("pant:datePublished")
     Field<Calendar> datePublished();
 
+    @Named("pant:dateFirstPublished")
```

Comment:
what's the difference between pant:datePublished and pant:dateFirstPublished?

Reply:
Great question. datePublished is specific to that particular version and is not carried forward to any new versions. dateFirstPublished is populated only once, on first-ever publish only, and then carried forward from version to version.

Reply:
Thank you for the info! how are we going to handle the modules/assemblies published in prod today that don't have pant:dateFirstPublished?

Reply:
Another good question. :)

I have the logic set up in AsciidoctorService.java:258 so that the 'datePublished' value is used for both the "updated" and "published" injected attributes. Then immediately after that, if the 'firstPublished' value exists, it overwrites the "published" injected attribute.

So all of our docs today will rely on the fallback logic. Over time it will become less necessary, but the bottom line is that today's docs will be handled as best as possible with what we have.